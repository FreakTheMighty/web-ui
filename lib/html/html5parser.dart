// Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.
// Generated by scripts/tree_gen.py.

/**
 * An HTML5 parser written in Dart.
 *
 * Intended to parse all html documents using the [html5 parsing algorithm][p],
 * although the implementation is not yet complete.
 *
 * [p]: http://dev.w3.org/html5/spec/parsing.html#parsing
 */
#library('html5parser');

#source('htmltree.dart');
#source('parser.dart');
#source('tree.dart');
#source('token.dart');
#source('tokenkind.dart');
#source('tokenizer_base.dart');
#source('tokenizer.dart');
