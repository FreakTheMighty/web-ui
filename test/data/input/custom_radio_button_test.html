<!doctype html>
<!--
Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
for details. All rights reserved. Use of this source code is governed by a
BSD-style license that can be found in the LICENSE file.
-->
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
  <title>watch_global_var_test</title>
  <script type='application/javascript' src="testing.js"></script>
</head>
<body>
  <element name="x-my-input" extends="input">
    <template></template>
    <script type="application/dart">
    import 'package:web_ui/web_ui.dart';
    class MyInput extends WebComponent {
      int clicked;
      created() {
        clicked = 0;
      }
      inserted() {
        on.click.add((e) { clicked++; });
      }
    }
    </script>
  </element>
  <form>
  <!-- TODO(jmesserly): if fix #82 we can use <x-my-input> tags. -->
  <input is="x-my-input" name="a" type="radio" value="Foo" bind-value="name">
  Foo!
  <input is="x-my-input" name="a" type="radio" value="Bar" bind-value="name">
  Bar!
  </form>
  <pre>You picked {{name}}. Final value should be 'Bar'</pre>
  <script type="application/dart">
    import 'dart:html';
    import 'package:web_ui/web_ui.dart';
    import 'package:unittest/unittest.dart';

    var name = 'Bar';
    var foo = queryAll('input')[0];
    var bar = queryAll('input')[1];

    main() {
      useShadowDom = false;

      expect(foo.checked, false, reason: 'watchers not run yet.');
      expect(bar.checked, false, reason: 'watchers not run yet.');

      window.setTimeout(() {
        expect(foo.checked, false, reason: 'bar picked.');
        expect(bar.checked, true, reason: 'bar picked.');
        expect(name, 'Bar');

        name = 'Nothing';
        dispatch();
        expect(foo.checked, false, reason: 'nothing picked.');
        expect(bar.checked, false, reason: 'nothing picked.');
        expect(name, 'Nothing');

        name = 'Foo';
        dispatch();
        expect(foo.checked, true, reason: 'foo picked.');
        expect(bar.checked, false, reason: 'foo picked.');
        expect(name, 'Foo');

        expect(foo.xtag.clicked, 0);
        expect(bar.xtag.clicked, 0);
        bar.on.click.dispatch(
            new MouseEvent('click', window, 1, 0, 0, 0, 0, 0));
        expect(name, 'Bar', reason: 'bar clicked.');
        expect(foo.checked, false, reason: 'bar clicked.');
        expect(bar.checked, true, reason: 'bar clicked.');
        expect(foo.xtag.clicked, 0);
        expect(bar.xtag.clicked, 1);

        foo.checked = true;
        expect(bar.checked, false, reason: 'only one can be checked.');
        expect(foo.checked, true, reason: 'explicitly set this.');
        // It would be nice if this was "Foo".
        expect(name, 'Bar', reason: 'DOM checked does not fire event.');

        window.setTimeout(() => window.postMessage('done', '*'), 0);
      }, 0);
    }
  </script>
</body>
</html>
