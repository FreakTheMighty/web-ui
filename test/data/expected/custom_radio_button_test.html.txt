Content-Type: text/plain
layer at (0,0) size 800x600
  RenderView at (0,0) size 800x600
layer at (0,0) size 800x71
  RenderBlock {HTML} at (0,0) size 800x71
    RenderBody {BODY} at (8,8) size 784x50
      RenderBlock {FORM} at (0,0) size 784x21
        RenderBlock {INPUT} at (5,3) size 13x13
        RenderText {#text} at (21,1) size 38x19
          text run at (21,1) width 4: " "
          text run at (25,1) width 34: "Foo! "
        RenderBlock {INPUT} at (64,3) size 13x13
        RenderText {#text} at (80,1) size 31x19
          text run at (80,1) width 4: " "
          text run at (84,1) width 27: "Bar!"
      RenderBlock {PRE} at (0,34) size 784x16
        RenderText {#text} at (0,0) size 88x16
          text run at (0,0) width 88: "You picked "
        RenderText {#text} at (88,0) size 24x16
          text run at (88,0) width 24: "Bar"
        RenderText {#text} at (112,0) size 232x16
          text run at (112,0) width 232: ". Final value should be 'Bar'"
